crates/auth/src/auth_req_token.rs

```diff
@@ -220,10 +221,11 @@ mod auth_middleware_tests {
             }),
             ..appstate::test_state().await
         };
+
         Router::new()
-            .route("/guarded", get(handler_return_200))
+            .route("/guarded", get(StatusCode::OK))
```

Comment:
Nice to see the handler impl for `T: IntoResponse + Clone` get some use! Any reason you didn't replace the last remaining use of `handler_return_200` as well?